Services/service_response.c

```diff
@@ -49,11 +49,22 @@ void service_response_task(void *param) {
     /* To get conn from the response queue */
     if (xQueueReceive(service_queues.response_queue, &packet,
                       NORMAL_TICKS_TO_WAIT) == pdPASS) {
-      cnv8_32(&packet->data[DATA_BYTE], &in);
-      printf("Set to %u\n", (uint32_t)in);
+      /* For some reason, if we directly print packet->data[DATA_BYTE],
+         it will be set to arg we sent.
+      */
+      printf("Set to %u\n",packet->data[DATA_BYTE]);
+      //cnv8_32(&packet->data[DATA_BYTE], &in);
+      //printf("Set to %u\n", (uint32_t)in);
+      
+      int res = csp_sendto(CSP_PRIO_NORM, packet->id.dst, packet->id.dport, packet->id.src, CSP_O_NONE, packet, 1000);
+      if (res != CSP_ERR_NONE) {
+        printf("Packet Sent back failed\n");
+      }else{
+        printf("Sent OK\n");
+      }
       csp_buffer_free(packet);
```

Comment:
@iBacklight You actually only need to free the backet if the `csp_sendto` failed